client/devdocs/design/search-collection.jsx

```diff
@@ -0,0 +1,97 @@
+/**
+* External dependencies
+*/
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+let FilterSummary, Hider;
+
+Hider = React.createClass( {
+	displayName: 'Hider',
+
+	propTypes: {
+		hide: React.PropTypes.bool,
+	},
+
+	shouldComponentUpdate( nextProps ) {
+		return this.props.hide !== nextProps.hide;
+	},
+
+	render() {
+		return (
+			<div style={ this.props.hide ? { display: 'none' } : {} }>
+				{ this.props.children }
+			</div>
+		);
+	}
+} );
+
+FilterSummary = React.createClass( {
+	render() {
+		var names;
+
+		if ( this.props.items.length === 0 ) {
+			return ( <p>No matches found</p> );
+		} else if ( this.props.items.length === this.props.total || this.props.items.length === 1 ) {
+			return null;
+		}
+
+		names = this.props.items.map( function( item ) {
+			return item.props.children.type.displayName;
+		} );
+
+		return (
+			<p>Showing: { names.join( ', ' ) }</p>
+		);
+	}
+} );
+
+export default React.createClass( {
+	displayName: 'Collection',
+
+	shouldWeHide: function( example ) {
+		var filter, searchString;
+
+		filter = this.props.filter || '';
+		searchString = example.type.displayName;
+
+		if ( this.props.component ) {
+			return example.type.displayName.toLowerCase() !== this.props.component.replace( /-([a-z])/g, '$1' );
+		}
+
+		if ( example.props.searchKeywords ) {
+			searchString += ' ' + example.props.searchKeywords;
+		}
+
+		return ! ( ! filter || searchString.toLowerCase().indexOf( filter ) > -1 );
+	},
+
+	visibleExamples: function( examples ) {
+		return examples.filter( ( child ) => {
+			return !child.props.hide;
+		} );
+	},
+
+	render: function() {
+		var summary, examples;
+
+		examples = this.props.children.map( ( example ) => {
```

Comment:
React discourages iterating over children like this, as it's not always reliable that the children are an array. Instead, they offer a helper to achieve this same purpose:

```jsx
examples = React.Children.map( this.props.children, ( example ) => {
```

More information: https://facebook.github.io/react/docs/top-level-api.html#react.children